src/traces/scattergl/index.js

```diff
@@ -104,14 +103,10 @@ function calc(gd, trace) {
     // and an average size for array marker.size inputs.
     if(count < TOO_MANY_POINTS) {
         ppad = calcMarkerSize(trace, count);
-        calcAxisExpansion(gd, trace, xa, ya, x, y, ppad);
-    } else {
-        if(markerOptions) {
-            ppad = 2 * (markerOptions.sizeAvg || Math.max(markerOptions.size, 3));
-        }
-        fastAxisExpand(xa, x, ppad);
-        fastAxisExpand(ya, y, ppad);
+    } else if(markerOptions) {
+        ppad = 2 * (markerOptions.sizeAvg || Math.max(markerOptions.size, 3));
     }
+    calcAxisExpansion(gd, trace, xa, ya, x, y, ppad);
```

Comment:
Very clean. This is great! 

To note though, this nice combined pathway is little slower at 1e6 pts. [Previously](https://github.com/plotly/plotly.js/pull/2404#issuecomment-368595753), `fastAxisExpand` would clock at ~5ms for one  coordinate, now we're more in a 20-30ms range. Even though this is significant, I think it's worthwhile to make scattergl autorange even more _on-par_ with the rest of the library. :+1: 

The failing tests are in this new [suite](https://github.com/plotly/plotly.js/blob/3bff26f7ef21c8ca43122e86214d268f57b77418/test/jasmine/tests/gl2d_plot_interact_test.js#L880) that got put in during https://github.com/plotly/plotly.js/pull/2404. Should be an easy fix.

Reply:
Huh, let's unpack that performance issue. Previously the inner loop was:
```js
for(var i = 0; i < vals.length; i += 2) {	
  var v = vals[i];	
  if(v < b0) b0 = v;	
  if(v > b1) b1 = v;	
}
```

this afternoon it was:
```js
for(i = 0; i < data.length; i++) {
  v = data[i];
  if(Math.abs(v) < FP_SAFE) {
    // data is not linearized yet so we still have to filter out negative logs
    if(v < vmin && (!isLog || v > 0)) vmin = v;
    if(v > vmax) vmax = v;
  }
}
```
The old code was accidentally omitting half the points (`i+= 2`), so we should be comparing 10ms to 20-30 ms (unless the 20-30ms is for both coordinates?). Then I'm doing two extra things: 1) comparing to `FP_SAFE` so we don't get a broken graph from numbers very close to the limit of floating point numbers, and 2) ensuring positive numbers when on a log scale. The new code also has a bunch more logic before and after this loop, but nothing else scales with the data size, all that is presumably <1ms. In my own tests don't see the extra logic *in* the loop coming anywhere near doubling the time. What am I missing?

Anyway I noticed I could optimize it a bit by making two versions of the loop, one for log and one for other types. Seems worth a couple of extra lines since this gets so much traffic -> https://github.com/plotly/plotly.js/pull/2437/commits/247c626a61a1fe813577f1e6fdf5fee9824680e6

Updated the failing test in https://github.com/plotly/plotly.js/pull/2437/commits/cf4842609e240a708cac093e07470bbaae3ca1a9

Reply:
Great! You're latest patch in 247c626 got `Axes.expand` within 5ms of the `fastAxisExpand` with `i++` swapped in for `i += 2` at 1e6 pts. Considering it's more robust and works with traces other than `scattergl`, I'll call this a win. :1st_place_medal: